src/main/java/org/truffleruby/language/exceptions/TopLevelRaiseHandler.java

```diff
@@ -80,4 +88,13 @@ public void setLastException(VirtualFrame frame, DynamicObject exception) {
         setExceptionVariableNode.setLastException(frame, exception);
     }
 
+    public void handleSignalException(DynamicObject exception) {
+        if (Layouts.BASIC_OBJECT.getLogicalClass(exception) == coreLibrary().getSignalExceptionClass()) {
+            getContext().getSafepointManager().pauseAllThreadsAndExecute(this, true, (thread, current_node) -> {
+                CallDispatchHeadNode node = CallDispatchHeadNode.getUncached();
+                node.call(exception, "reached_top_level");
+            });
```

Comment:
This should likely just be called on the current thread, why is a safepoint used?

Reply:
On MRI this happens during VM tear down when no ruby thread is executing.  So I thought I would emulate that here.

Reply:
Right, but we're already pretty much in tear down code here so it seems unnecessary.

BTW, I just noticed that if the SignalException kills a thread, no native signal is sent, which is already the behavior as TopLevelRaiseHandler is only used for the main thread :)